src/lang/std/artifactGraph.ts

```diff
@@ -871,3 +871,15 @@ export function codeRefFromRange(range: SourceRange, ast: Program): CodeRef {
     pathToNode: getNodePathFromSourceRange(ast, range),
   }
 }
+
+export function isSolid2D(artifact: Artifact): artifact is solid2D {
+  return (artifact as solid2D).pathId !== undefined
+}
+
+export function isSegment(artifact: Artifact): artifact is SegmentArtifact {
+  return (artifact as SegmentArtifact).pathId !== undefined
+}
+
+export function isSweep(artifact: Artifact): artifact is SweepArtifact {
+  return (artifact as SweepArtifact).pathId !== undefined
+}
```

Comment:
Sorry that I'm late on this @nadr0 .

What's up with these functions? I don't get how they're asserting what they're supposed to? we're still forcing the type with `as`. Why not use the type property, doing so has the typescript understanding the type narrowing, i.e. see my IDE knows what's up with no `as` coercion

![image](https://github.com/user-attachments/assets/22e80b92-824d-4e24-b018-e35fdb4e5c89)

Not so for checking the path id

![image](https://github.com/user-attachments/assets/472bdd38-d7e3-4a13-8970-486c202408c8)


In general I don't understand why all three assert the exact same thing.

Even if you did `pathId` in artifact, you would get a union type of artifacts with that property

![image](https://github.com/user-attachments/assets/8cf4ff7e-3eab-447d-9c99-747e94f8c2ff)

Anyway I think these should probably be 
```ts
export function isSolid2D(artifact: Artifact): artifact is Solid2DArtifact {
  return artifact.type === 'solid2D'
}

export function isSegment(artifact: Artifact): artifact is SegmentArtifact {
  return artifact.type === 'segment'
}

export function isSweep(artifact: Artifact): artifact is SweepArtifact {
  return artifact.type === 'sweep'
}
```

Reply:
Sorry these were really triggering me, they're a foot gun waiting to happen if someone else tries to use the utils and things that they are actually doing type narrowing when in fact they are just forcing the type.

I had a look at where they were used and decided to make a PR.
https://github.com/KittyCAD/modeling-app/pull/4759